fe/src/main/java/org/apache/doris/plugin/PluginMgr.java

```diff
@@ -76,14 +87,21 @@ public PluginInfo installPlugin(InstallPluginStmt stmt) throws IOException, User
         try {
             PluginInfo info = pluginLoader.getPluginInfo();
 
-            if (plugins[info.getTypeId()].putIfAbsent(info.getName(), pluginLoader) != null) {
-                pluginLoader.uninstall();
+            if (plugins[info.getTypeId()].containsKey(info.getName())) {
                 throw new UserException("plugin " + info.getName() + " has already been installed.");
             }
-
+            
             // install plugin
             pluginLoader.install();
             pluginLoader.setStatus(PluginStatus.INSTALLED);
+            
+            if (plugins[info.getTypeId()].putIfAbsent(info.getName(), pluginLoader) != null) {
+                pluginLoader.uninstall();
```

Comment:
We can't uninstall the plugin here, because it may cause the actual plugin to be uninstalled by mistake